### PR TITLE
Topic information

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connects to ros2/ros1_bridge#3

The `rmw_connext_cpp` implementation is ready for review.

### CODE REVIEW

File: rmw_connext_cpp/src/functions.cpp [hunk 1]
```diff
@@ -1630,4 +1846,246 @@ rmw_send_response(
...
+  }
+  if (!node_info->subscriber_listener) {
+    RMW_SET_ERROR_MSG("subscriber listener handle is null");
+    return RMW_RET_ERROR;
+  }
+
+  // combine publisher and subscriber information
+  std::map<std::string, std::set<std::string>> topics_with_multiple_types;
+  for (auto it : node_info->publisher_listener->topic_names_and_types) {
+    for (auto jt : it.second) {
```
Comment (wjwwood) [inline rmw_connext_cpp/src/functions.cpp]: Maybe use `auto & jt` here to avoid copying the string.

File: rmw_connext_cpp/src/functions.cpp [hunk 2]
```diff
@@ -1630,4 +1846,246 @@ rmw_send_response(
...
+
+  // reformat type name
+  std::string substr = "::msg::dds_::";
+  for (auto it : topics) {
+    size_t substr_pos = it.second.find(substr);
+    if (it.second[it.second.size() - 1] == '_' && substr_pos != std::string::npos)
+    {
+      it.second = it.second.substr(0, substr_pos) + "/" +
+        it.second.substr(substr_pos + substr.size(), it.second.size() - substr_pos - substr.size() - 1);
+      topics[it.first] = it.second;
```
Comment (wjwwood) [inline rmw_connext_cpp/src/functions.cpp]: I don't think this should be necessary if you get a reference to the pair from the `topics` map.

**Comment by wjwwood** [comment]
lgtm

File: rmw_connext_cpp/src/functions.cpp (hunk 1 above)
Comment (dirk-thomas) [inline rmw_connext_cpp/src/functions.cpp]: Good point. Done in 04cef82c99e31c35cd58d0786d422ccca15ccb98

File: rmw_connext_cpp/src/functions.cpp (hunk 2 above)
Comment (dirk-thomas) [inline rmw_connext_cpp/src/functions.cpp]: Also done in 04cef82c99e31c35cd58d0786d422ccca15ccb98

**Comment by tfoote** [comment]
+1

**Comment by esteve** [comment]
+1

**Comment by esteve** [comment]
BTW, here's an idea, we could expose this information as a parameter too. I haven't given much thought to it, though.

**Pull request merged**